include/mqt-core/dd/GateMatrixDefinitions.hpp

```diff
@@ -174,4 +140,15 @@ inline TwoQubitGateMatrix xxPlusYYMat(const fp theta, const fp beta = 0.) {
        {0, {-sinBeta * sinTheta, -cosBeta * sinTheta}, cosTheta, 0},
        {0, 0, 0, 1}}};
 }
+
+inline GateMatrix opToSingleGateMatrix(qc::OpType t,
+                                       const std::vector<fp>& params = {}) {
+  return MATS_GENERATORS.at(t)(params);
+}
+
+inline TwoQubitGateMatrix
+opToTwoQubitGateMatrix(qc::OpType t, const std::vector<fp>& params = {}) {
+  return TWO_GATE_MATS_GENERATORS.at(t)(params);
+}
```

Comment:
This is more of a general comment/question:
Given the presence of these functions, are the factory functions actually needed/helpful? 
Don't they simply add an additional level of indirection (if not two, based on the use of lambdas)?
Wouldn't it be enough to simply implement these two methods and have them use an internal switch statement on the OpType?
From experience, the compiler generates pretty efficient code from switch statements that is actually very close to a LUT.

Reply:
I'm not sure which one is more efficient, because the switch (for the compiler) is non exhaustive, which requires it to generate some checks. But i think the performance differences are minimal, if there are any. But i can rewrite it to use switch statements.

Reply:
I believe rewriting to switch statements could reduce the overall complexity of the code at (at most) a negligible runtime overhead. So I'd be in favor of doing that if there are no strong objections.